mp4parse/src/lib.rs

```diff
@@ -2003,12 +2004,10 @@ fn read_ipma<T: Read>(
         };
 
         if let Some(previous_association) = associations.last() {
-            if previous_association.item_id > item_id {
-                return Err(Error::InvalidData(
-                    "Each ItemPropertyAssociation box shall be ordered by increasing item_ID",
-                ));
-            } else if previous_association.item_id == item_id {
-                return Err(Error::InvalidData("There shall be at most one association box for each item_ID, in any ItemPropertyAssociation box"));
+            match previous_association.item_id.cmp(&item_id) {
+                Ordering::Greater => return Err(Error::InvalidData( "Each ItemPropertyAssociation box shall be ordered by increasing item_ID")),
+                Ordering::Equal => return Err(Error::InvalidData("There shall be at most one association box for each item_ID, in any ItemPropertyAssociation box")),
+                Ordering::Less => {}
```

Comment:
Personally, I find this less readable than the if/else version. I'd prefer adding an local exception or globally allowing this with a `-A clippy::comparison-chain` option. That said, if you feel strongly, I'm not about to die on this hill or anything.

Reply:
I agree with you, I'll add an allow line for this instead.